libraries/AP_DroneCAN/AP_Canard_iface.cpp

```diff
@@ -214,6 +214,17 @@ void CanardInterface::processTx(bool raw_commands_only = false) {
         if (txq == nullptr) {
             return;
         }
+        const auto *stats = ifaces[iface]->get_statistics();
+        bool iface_down = true;
+        if (stats == nullptr || (AP_HAL::micros64() - stats->last_transmit_us) < 200000UL) {
```

Comment:
i think we need to fetch the last_transmit_us into a variable first, then get the micros64() into a variable, then subtract
the issue is that if the CAN thread updates last_transmit_us after we do the micros64() call then we will get a very large number, and will think the interface is down, so may lose a packet

Reply:
Correct, I also marked *stats as volatile so compiler doesn't try to optimise it.

Reply:
Also the issue was present in the AP_Periph implementation as well, so fixed there as well.